### PR TITLE
feat(epic-i): Implement Phase I-3 Autonomous Evolution with benchmark evaluation and capability scoring

### PR DESCRIPTION
**Pull request by RC918**
# feat(epic-i): Implement Phase I-3 Autonomous Evolution with benchmark evaluation

## Summary

This PR implements EPIC I Phase I-3 (Autonomous Evolution) which adds periodic benchmark evaluation and capability scoring for LLM providers. The system can now self-evaluate provider performance and automatically flag degrading providers for review.

**New Files:**
- `benchmark_evaluator.py` - BenchmarkTask schema, BenchmarkEvaluator class with predefined benchmark suite (code generation, code review, bug fix tasks)
- `capability_score_manager.py` - ProviderCapabilityScore with trend analysis (improving/stable/degrading) and auto-update rules

**Key Features:**
- Scheduled benchmarks: Weekly full suite (Sunday 02:00 UTC), Daily smoke tests (02:00 UTC)
- Auto-update rules: >10% drop triggers alert, >20% drop triggers auto-downgrade
- Safety guard: No auto-upgrade without human review
- Dry-run mode enabled by default (simulates results without actual LLM calls)

**Feature Flags:**
- `BENCHMARK_EVALUATION_ENABLED` - Enable/disable benchmark evaluation (default: false)
- `CAPABILITY_SCORE_ENABLED` - Enable/disable capability scoring (default: false)

## Review & Testing Checklist for Human

- [ ] **Verify auto-update thresholds** (lines 377-399 in capability_score_manager.py): >10% drop = alert, >20% drop = auto-downgrade. Confirm these align with EPIC I-3 roadmap requirements.
- [ ] **Review hardcoded provider/model lists** (lines 600-607, 639-643 in heartbeat_handler.py): These are hardcoded for weekly/daily benchmarks. Consider if this should be configurable.
- [ ] **Test schedule timing logic** (lines 529-548 in heartbeat_handler.py): The weekday/hour checks for benchmark scheduling may have edge cases around timezone boundaries.
- [ ] **Verify dry-run simulation scores** (lines 451-489 in benchmark_evaluator.py): The simulated scores vary by provider - confirm these are reasonable for testing.

**Recommended Test Plan:**
1. Enable feature flags in staging: `BENCHMARK_EVALUATION_ENABLED=true`, `CAPABILITY_SCORE_ENABLED=true`
2. Manually trigger a benchmark run by temporarily adjusting the schedule check
3. Verify capability scores are stored in Redis and alerts are generated for score changes
4. Confirm no impact on production routing (benchmark failures should be isolated)

### Notes

Blueprint Alignment: Section 4.3 (Model Governance Framework v2), Section 4.4 (Autonomous Provisioning v2)

---
Requested by: @RC918
Link to Devin run: https://app.devin.ai/sessions/b08371b5aaa34b1a84cc84a1647ecd54

---

> [!NOTE]
> Implements EPIC I-3 autonomous evolution by adding benchmark evaluation and capability scoring, wired into the governance heartbeat with strict safety guards.
> 
> - New `governance/benchmark_evaluator.py`: defines `BenchmarkTask`/`BenchmarkResult`, a predefined suite (code gen/review/bug-fix), weighted scoring, dry-run simulation, and Redis result storage (`governance:benchmark_results:*`).
> - New `governance/capability_score_manager.py`: EMA-based provider/model scores with trends (improving/stable/degrading), Redis persistence/history/alerts, and change rules (>10% flag, >20% auto-downgrade; never auto-upgrade).
> - Heartbeat updates (`governance/heartbeat_handler.py`): adds `_should_run_benchmark` and `_run_benchmark_evaluation`; schedules weekly full (Sun 02:00 UTC) and daily smoke (02:00 UTC), updates scores post-run, and records last-run keys.
> - Safety/ops: failures isolated from routing, non-blocking lock retained; controlled via `BENCHMARK_EVALUATION_ENABLED`, `BENCHMARK_DRY_RUN`, `CAPABILITY_SCORE_ENABLED`, `CAPABILITY_SCORE_DRY_RUN`.
> 
> <sup>Written by [Cursor Bugbot](https://cursor.com/dashboard?tab=bugbot) for commit a34ad515d28999cdf1f7d1a9f86b91511bc3be52. Configure [here](https://cursor.com/dashboard?tab=bugbot).</sup>

### CODE REVIEW

File: handoff/20250928/40_App/orchestrator/governance/benchmark_evaluator.py [hunk 1]
```diff
@@ -0,0 +1,693 @@
...
+            from llm.client import get_llm_client
+
+            client = get_llm_client()
+
+            # Execute the benchmark task
+            response = client.generate(
+                prompt=task.prompt,
+                provider=provider,
+                model=model,
+                timeout=task.timeout_seconds,
```
Comment (RC918) [inline handoff/20250928/40_App/orchestrator/governance/benchmark_evaluator.py]: IMPACT: The timeout parameter is directly passed from the task configuration without validation. If the task's timeout is set to an unreasonable value, it could lead to long-running processes that impact system stability.

SUGGESTED FIX: """
# Add reasonable timeout limits
MAX_TIMEOUT_SECONDS = 120
timeout = min(task.timeout_seconds, MAX_TIMEOUT_SECONDS)
"""
